### DISCUSSION TITLE
terminal throws error when trying to see forge version...

### DISCUSSION THREAD

**Discussion by Parham0x79** [Cyfrin/foundry-full-course-cu#1667] [Q&A] [answered]
![Screenshot (2)](https://github.com/Cyfrin/foundry-full-course-f23/assets/163135012/76e209ab-f8a8-4f91-b732-872a0c3f41ec)
so i'm getting this error while trying to see forge version.... what's the problem?
I already asked chatGPT and it said something about compatibility problems which didn't really understood!

**Comment by EngrPips** [comment]
Hey @Parham0x79, Are you using `wsl` ?

**Reply by Parham0x79** [reply]
yes! i installed ubuntu subsystem it's working perfectly fine.

**Reply by EngrPips** [reply]
Awesome, I am so glad to hear you figured it out.

**Reply by Parham0x79** [reply]
Noooo! LOL !! there's a misundresting here!
i just said the wsl command on the terminal is working ... my main issue is with the forge which doesn't give me the version

**Reply by Parham0x79** [reply]
I asked ChatGPT this: 
![Screenshot (3)](https://github.com/Cyfrin/foundry-full-course-f23/assets/163135012/aeccd5b7-48f6-4bd4-b07b-ef402a1c91ef)

and this is the answer i received which i don't undrestand:
![Screenshot (4)](https://github.com/Cyfrin/foundry-full-course-f23/assets/163135012/1b9c9091-793b-458c-8597-dbf243f40183)

**Reply by EngrPips** [reply, marked as answer]
This is a long list of solutions to try out. You may have downloaded the wrong foundry binary for your system. You should re-install Foundry. You can follow the steps in the Foundry Book https://book.getfoundry.sh/getting-started/installation.

**Reply by Parham0x79** [reply]
followed the instructions as was mentioned in the document and installed rust and cargo successfully (which took almost 3 hours!)
I can get the version of "cast, chisel and anvil" but still returns the same error for forge version
```
parham0x79@DESKTOP-3EIJDAE:~$ forge --version
bash: /home/parham0x79/.foundry/bin/forge: cannot execute binary file: Exec format error
parham0x79@DESKTOP-3EIJDAE:~$ cast --version
cast 0.2.0 (5885dbc3 2024-05-01T11:31:14.077096700Z)
parham0x79@DESKTOP-3EIJDAE:~$ chisel --version
chisel 0.2.0 (5885dbc3 2024-05-01T11:52:07.697192500Z)
parham0x79@DESKTOP-3EIJDAE:~$ anvil --version
anvil 0.2.0 (5885dbc3 2024-05-01T12:14:32.991599600Z)
parham0x79@DESKTOP-3EIJDAE:~$ 
```

**Reply by EngrPips** [reply]
It seems like the Forge binary is precisely what the problem is. Do you have issues whenever you run any other forge command?

**Reply by Parham0x79** [reply]
it worked.....! after I opened a new terminal finaaaally!!!! 
i still didn't get what the problem was but huge thanks to you i couldn't figure it out without your help 
what a struggle...LOL!!!

**Reply by EngrPips** [reply]
I am delighted to hear that you got it to work. What a struggle, lol.
